data-node/src/main/java/org/graylog/datanode/rest/StatusController.java

```diff
@@ -48,4 +49,9 @@ public DataNodeStatus status() {
                 new StatusResponse(datanodeConfiguration.opensearchDistributionProvider().get().version(), openSearch.processInfo())
         );
     }
+
+    @DELETE
+    public void remove() {
```

Comment:
I'd suggest moving this to a different controller (management?), to keep the operations separated. 

The management controller probably shouldn't be accessible if the datanode is started in an insecure mode and there is no auth that would guard these, potentially very dangerous, actions.

Reply:
I agree

Reply:
Good point. I only added the endpoint to be able to trigger removal from the integration test. 
I'll move it to a separate controller and secure it with an annotation